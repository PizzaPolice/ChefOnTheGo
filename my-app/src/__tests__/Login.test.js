import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import Login from '../components/login.js'

//This test is shallow rendering our Login component
describe('Login Component', () => {
    it('Login should render without throwing an error', () => {
        expect(shallow(<Login />).exists(<form className='Login'></form>)).toBe(true)
      })
      it("should render initial layout", () => {
        const component = shallow(<Login />);
        expect(component.getElements()).toMatchSnapshot();
        })
    });

    describe('Login Button', () => {
        it('should be defined', () => {
          expect(Login).toBeDefined();
        });

        it('should render correctly', () => {
          const button = shallow(
            <Login name='btn next-btn' />
          );
          expect(button).toMatchSnapshot();
        });
       });


/*
it('Login component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Login />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
*/
