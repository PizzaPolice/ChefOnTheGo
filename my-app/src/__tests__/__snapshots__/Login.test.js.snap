// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render initial layout 1`] = `
Array [
  <div
    className="outer-panel"
>
    <div
        className="panel-body"
    >
        <h2
            className="title"
        >
            Login
        </h2>
        <div
            className="input-fields"
        >
            <div
                className="input-line"
            >
                Username:
                <input
                    name="username"
                    onChange={[Function]}
                    type="text"
                />
            </div>
            <div
                className="input-line"
            >
                Password:
                <input
                    name="password"
                    onChange={[Function]}
                    type="password"
                />
            </div>
            <div
                className="radio-input"
            >
                <label
                    htmlFor="customer"
                >
                    Customer
                </label>
                <input
                    id="customer"
                    name="login_type"
                    onChange={[Function]}
                    type="radio"
                    value="customer"
                />
                <label
                    htmlFor="chef"
                >
                    Chef
                </label>
                <input
                    id="chef"
                    name="login_type"
                    onChange={[Function]}
                    type="radio"
                    value="chef"
                />
            </div>
        </div>
    </div>
    <div
        className="panel-footer"
    >
        <Link
            replace={false}
            to="signup"
        >
            <button
                className="btn back-btn"
            >
                New User?
            </button>
        </Link>
        <Link
            onClick={[Function]}
            replace={false}
            to="login"
        >
            <button
                className="btn next-btn"
            >
                Login
            </button>
        </Link>
    </div>
</div>,
]
`;
